code/modules/mob/mob_movement.dm

```diff
@@ -152,146 +157,135 @@
 		if(mob.hud_used)
 			mob.hud_used.show_hud(HUD_STYLE_STANDARD)
 
-	if(mob.control_object)	Move_object(direct)
+	if(world.time < move_delay)
+		return
 
-	if(world.time < move_delay)  return
+	move_delay = world.time + world.tick_lag //this is here because Move() can now be called multiple times per tick
+	if(!mob || !mob.loc)
+		return 0
 
-	if(!isliving(mob))  return mob.Move(n,direct)
+	if(mob.notransform)
+		return 0 //This is sota the goto stop mobs from moving var
 
-	if(moving)  return 0
+	if(mob.control_object)
+		return Move_object(direct)
 
-	if(!mob)	return
+	if(!isliving(mob))
+		return mob.Move(n, direct)
 
-	if(mob.stat==DEAD)	return
+	if(mob.stat == DEAD)
+		mob.ghostize()
+		return 0
 
-	if(mob.notransform)	return//This is sota the goto stop mobs from moving var
+	if(moving)
+		return 0
 
 	if(isliving(mob))
 		var/mob/living/L = mob
 		if(L.incorporeal_move)//Move though walls
 			Process_Incorpmove(direct)
 			return
 
-	if(Process_Grab())	return
-
-	if(mob.buckled)							//if we're buckled to something, tell it we moved.
-		return mob.buckled.relaymove(mob, direct)
-
-	if(mob.remote_control)					//we're controlling something, our movement is relayed to it
+	if(mob.remote_control) //we're controlling something, our movement is relayed to it
 		return mob.remote_control.relaymove(mob, direct)
 
 	if(isAI(mob))
 		if(istype(mob.loc, /obj/item/device/aicard))
 			var/obj/O = mob.loc
-			return O.relaymove(mob, direct) //aicards have special relaymove stuff
-		return AIMove(n,direct,mob)
+			return O.relaymove(mob, direct) // aicards have special relaymove stuff
+		return AIMove(n, direct, mob)

```

Comment:
I wonder if this is more OOP on TG; they don't have the aicard typecheck; wonder if that's handled in their `AIMove`?